packages/connect-wallet-modal/src/components/Ledger/LedgerContext.tsx

```diff
@@ -0,0 +1,109 @@
+import React, {
+  createContext,
+  FC,
+  MutableRefObject,
+  useCallback,
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+} from 'react';
+import type Transport from '@ledgerhq/hw-transport';
+import Eth from '@ledgerhq/hw-app-eth';
+import { getTransport, interceptLedgerError } from './helpers';
+
+export interface LedgerContextProps {
+  isActive: boolean;
+}
+
+export type LedgerContextValue = {
+  transport: MutableRefObject<Transport | null>;
+  ledgerAppEth: MutableRefObject<Eth | null>;
+  isTransportConnected: boolean;
+  error: Error | null;
+  setError: (e: Error | null) => void;
+  connectTransport: () => Promise<void>;
+  disconnectTransport: () => Promise<void>;
+  reconnectTransport: () => Promise<void>;
+};
+
+export const LedgerContext = createContext({} as LedgerContextValue);
+
+export const LedgerContextProvider: FC<LedgerContextProps> = ({
+  isActive,
+  children,
+}) => {
+  const transport = useRef<Transport | null>(null);
+  const ledgerAppEth = useRef<Eth | null>(null);
+  const [error, setError] = useState<Error | null>(null);
+  const [isTransportConnected, setIsTransportConnected] = useState(false);
+  const [activeAccountsRequestsCounter, setActiveAccountsRequestsCounter] =
+    useState(0);
+
+  const disconnectTransport = useCallback(async (goingToReconnect = false) => {
+    await transport?.current?.close();
```

Comment:
can it throw? should try-catch handling be here?

Reply:
I would say, everything below should actually be within `finally {}` block

Reply:
I will just wrap everything in `try catch`, without `finally`. Because, if the transport wasn't able to `close()`, then it is a bad idea to reset it to null – the device will get stuck in the opened state until page reload.